prov/gni/src/gnix_cq.c

```diff
@@ -282,12 +282,29 @@ static int __gnix_cq_progress(struct gnix_fid_cq *cq)
 /*******************************************************************************
  * Exposed helper functions
  ******************************************************************************/
-ssize_t _gnix_cq_add_event(struct gnix_fid_cq *cq, void *op_context,
-			   uint64_t flags, size_t len, void *buf,
-			   uint64_t data, uint64_t tag, fi_addr_t src_addr)
+ssize_t _gnix_cq_add_event(struct gnix_fid_cq *cq, struct gnix_fid_ep *ep,
+                           void *op_context, uint64_t flags, size_t len,
+                           void *buf, uint64_t data, uint64_t tag,
+                           fi_addr_t src_addr)
 {
 	struct gnix_cq_entry *event;
 	struct slist_entry *item;
+        uint64_t op_flags;
+
+        if (ep) {
+                op_flags = ep->op_flags;
+
+                if (op_flags & FI_NOTIFY_FLAGS_ONLY) {
+                        uint64_t mask = (FI_REMOTE_CQ_DATA | FI_MULTI_RECV);
```

Comment:
also, doesn't need to be part of this PR, but for KNL optimizations this function should be moved to gni_cq.h and made static/inlineable